### PR TITLE
feat: add capability of creating ingresses

### PR DESCRIPTION
**Pull request by smuu**
Resolves https://github.com/celestiaorg/knuu/issues/121

Tested here: https://github.com/celestiaorg/knuu-example/blob/ff383fe0820f138f0ce4f7946d0835256087bfca/basic/ingress_test.go

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

**Review by jrmanes — APPROVED** [review]
LGTM! great job! 🚀 
also, thanks for the TODOs, these notes are very useful for future interactions

File: pkg/knuu/instance_helper.go [hunk 1]
```diff
@@ -314,3 +314,45 @@ func (i *Instance) addFileToBuilder(src string, dest string, chown string) error
...
+	ingress := i.ingress
+
+	annotations := map[string]string{}
+
+	if ingress.BackendProtocol != "" {
+		annotations["nginx.ingress.kubernetes.io/backend-protocol"] = ingress.BackendProtocol
+	}
+
+	if ingress.CertManagerEnabled {
+		//annotations["kubernetes.io/tls-acme"] = "true"
```
Comment (Bidon15) [inline pkg/knuu/instance_helper.go:329]: please explain why we need this comment out

**Review by Bidon15 — CHANGES_REQUESTED** [review]

File: pkg/k8s/k8s_ingress.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+	var tls []v1.IngressTLS
+	if tlsEnabled {
+		tls = []v1.IngressTLS{
+			{
+				Hosts:      []string{host},
+				SecretName: fmt.Sprintf("%s-tls", name),
+			},
+		}
+	}
```
[the remark this answers is not in the document]
Comment (smuu) [reply pkg/k8s/k8s_ingress.go:22]: The developer might want to deploy without TLS for testing something.

File: pkg/k8s/k8s_ingress.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+import (
+	"context"
+	"fmt"
+	v1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"time"
+)
+
+// CreateIngress creates a new Ingress resource.
+func CreateIngress(namespace string, name string, labels map[string]string, annotations map[string]string, ingressClassName string, host string, path string, pathType string, serviceName string, servicePort int, tlsEnabled bool) error {
```
[the remark this answers is not in the document]
Comment (smuu) [reply pkg/k8s/k8s_ingress.go]: Thank you :+1:

File: pkg/k8s/k8s_ingress.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (smuu) [reply pkg/k8s/k8s_ingress.go:21]: Agreed with @jrmanes to not fail when TLS is not enabled.

File: pkg/knuu/instance_helper.go (hunk 1 above)
Comment (smuu) [reply pkg/knuu/instance_helper.go]: This was a leftover from testing. We have this annotation in our kustomize manifests files, but the value is deprecated.
I removed this comment in: https://github.com/celestiaorg/knuu/pull/151/commits/2035aa15f62223ca002f5311c14c2493858854bc

**Pull request merged**